src/components/ConfirmedRoute.tsx

```diff
@@ -1,64 +1,54 @@
-import lodashGet from 'lodash/get';
-import lodashIsNil from 'lodash/isNil';
-import PropTypes from 'prop-types';
 import React, {useCallback, useEffect} from 'react';
+import {ImageSourcePropType} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import {OnyxEntry} from 'react-native-onyx/lib/types';
+import {SvgProps} from 'react-native-svg';
 import useNetwork from '@hooks/useNetwork';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as TransactionUtils from '@libs/TransactionUtils';
 import * as MapboxToken from '@userActions/MapboxToken';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import {MapboxAccessToken, Transaction} from '@src/types/onyx';
+import {WaypointCollection} from '@src/types/onyx/Transaction';
 import DistanceMapView from './DistanceMapView';
 import * as Expensicons from './Icon/Expensicons';
 import ImageSVG from './ImageSVG';
+import {WayPoint} from './MapView/MapViewTypes';
 import PendingMapView from './MapView/PendingMapView';
-import transactionPropTypes from './transactionPropTypes';
-
-const propTypes = {
-    /** Transaction that stores the distance request data */
-    transaction: transactionPropTypes,
 
+type ConfirmedRoutePropsOnyxProps = {
     /** Data about Mapbox token for calling Mapbox API */
-    mapboxAccessToken: PropTypes.shape({
-        /** Temporary token for Mapbox API */
-        token: PropTypes.string,
-
-        /** Time when the token will expire in ISO 8601 */
-        expiration: PropTypes.string,
-    }),
+    mapboxAccessToken: OnyxEntry<MapboxAccessToken>;
 };
 
-const defaultProps = {
-    transaction: {},
-    mapboxAccessToken: {
-        token: '',
-    },
+type ConfirmedRouteProps = ConfirmedRoutePropsOnyxProps & {
+    /** Transaction that stores the distance request data */
+    transaction: Transaction;
 };
 
-function ConfirmedRoute({mapboxAccessToken, transaction}) {
+function ConfirmedRoute({mapboxAccessToken, transaction}: ConfirmedRouteProps) {
     const {isOffline} = useNetwork();
-    const {route0: route} = transaction.routes || {};
-    const waypoints = lodashGet(transaction, 'comment.waypoints', {});
-    const coordinates = lodashGet(route, 'geometry.coordinates', []);
+    const route = transaction.routes?.route0 ?? {geometry: {coordinates: []}};
```

Comment:
`{geometry: {coordinates: []}}` This can be a breaking change, please test it carefully